### ISSUE TITLE
Add the option "--locked" to osdk for more deterministic build

### ISSUE THREAD

**Issue by kylerky** [asterinas/asterinas#1679]
### Feature Description

As described in the title

### Problem or Need

Cargo will bump the versions of the dependencies of asterinas, resulting in breakage of the build of the kernel. Many of the third party dependencies are still in the 0.x.y phase and introduce breaking changes even when bumping the patch versions. Asterinas has at least encountered such problem twice (see 6d3bb5a9d0b1d4d53e298402bc8fec57dc899f15 and 5f2bd9d0ac6b93a2dfe6e47da7ade2dbcbd57bb6), where we have to update the version of the toolchain.

### Suggested Implementation

Introduce "--locked" to the common options of the osdk actions.

### Possible Alternatives

Provide third party dependencies in Asterinas' GitHub

### Additional Context and Screenshots

Related: https://github.com/asterinas/asterinas/issues/1640

**Comment by tangruize** [comment]
I'm also adding an `--offline` option (#1673), which may address the same problem. I prefer `--offline` because `--locked` still requires network access and will reject to build if `Cargo.lock` needs updating. We can consider using both `--offline` and `--locked`, which is equivalent to `--frozen`.

**Comment by kylerky** [comment]
You are right in that `--locked` is not enough. I do agree that `--offline` is the way to go.
